term/src/terminalstate/sixel.rs

```diff
@@ -123,9 +131,10 @@ impl TerminalState {
             z_index: 0,
             image_id: None,
             placement_id: None,
-            do_not_move_cursor: false,
+            do_not_move_cursor: self.sixel_display_mode,
         }) {
             log::error!("set sixel image: {:#}", err);
         }
+        self.cursor = old_cursor;
```

Comment:
should this be conditional on sixel_display_mode?  Feels like we're going to revert the cursor position in the case where we did want it to move?

Reply:
Up to you.  old_cursor is only changed for sixel_display_mode.  There is also sixel_scrolls_right mode which I would assume DECSDM overrides based on ctlseqs: "Sixel scrolling leaves cursor to right of graphic." ; and that is dealt with in assign_image_to_cells.

The abstraction is leaking. Do we want a single place to choose both start and end position?  That would put it in assign_image_to_cells; but then we're putting a really sixel-only-ish logic in there.  Would it be better to make bottom_right a parameter from outside assign_image_to_cells so that scrolling and DECSDM are handled together in sixel() ?

```
       if !params.do_not_move_cursor {
            // Sixel places the cursor under the left corner of the image,
            // unless sixel_scrolls_right is enabled.
            // iTerm places it after the bottom right corner.
            let bottom_right = match params.style {
                ImageAttachStyle::Kitty | ImageAttachStyle::Iterm => true,
                ImageAttachStyle::Sixel => self.sixel_scrolls_right,
            };

            if bottom_right {
                self.set_cursor_pos(
                    &Position::Relative(width_in_cells as i64),
                    &Position::Relative(-1),
                );
            }
        }
```

Reply:
Re-reading that made me think that we should just put start/end position as parameters to assign_image_to_cells, and make all of the image formats pass it in.  Because that's really what this is: they have their own rules, and shouldn't be mixed up.  assign_image_to_cells should just be slice-and-dice: then it could be re-used anywhere else you might want to splat a bitmap on screen later on.

Reply:
what I meant was: this appears to unconditionally reset `self.cursor` to old cursor, which would undo any cursor movement that might have been done by `assign_image_to_cells` in the `do_not_move_cursor: false` case.

Reply:
Oops!  You're right.  I didn't see it because I never allow sixel to scroll.